### PR TITLE
[tests-only] TUS tests for checksum

### PR DESCRIPTION
**Pull request by swoichha**
## Description
This PR adds test for TUS checksum.

## Related Issue
- https://github.com/owncloud/ocis/issues/1599

## How Has This Been Tested?
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/acceptance/features/bootstrap/TUSContext.php [hunk 1]
```diff
@@ -327,4 +329,25 @@ public function userUploadsWithCreatesWithUpload(
 		);
 		\unlink($tmpfname);
 	}
+
+	/**
+	 * @When user :user uploads file with checksum :checksum to the last created TUS Location with offset :offset and content :content using the TUS protocol on the WebDAV API
+	 * @Given user :user has uploaded file with checksum :checksum to the last created TUS Location with offset :offset and content :content using the TUS protocol on the WebDAV API
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TUSContext.php:335]: The Given step seems not to be used, if you want to have a given step, please split it off and add at least simple checks to see if it succeeded

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/bootstrap/TUSContext.php (hunk 1 above)
Comment (swoichha) [reply tests/acceptance/features/bootstrap/TUSContext.php:335]: It has been used here https://github.com/owncloud/core/pull/38466/files#diff-26b35f83f03795db544087e974a4b86d10d40038b2f5183c82df603984f08301R40

File: tests/acceptance/features/bootstrap/TUSContext.php [hunk 2]
```diff
@@ -327,4 +329,43 @@ public function userUploadsWithCreatesWithUpload(
...
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userHasUploadedFileWithChecksum(
+		string $user,
+		string $checksum,
+		string $offset,
+		string $content
+	) {
+		$this->sendsAChunkToTUSLocationWithOffsetAndData($user, $offset, $content, $checksum);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TUSContext.php:369]: need a check here if the step actually succeeded

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

**Review by individual-it — APPROVED** [review]

**Pull request merged**